extensions/arrow/src/main/java/io/deephaven/extensions/arrow/sources/ArrowBooleanColumnSource.java

```diff
@@ -0,0 +1,53 @@
+/*
+ * Copyright (c) 2016-2023 Deephaven Data Labs and Patent Pending
+ */
+/*
+ * ---------------------------------------------------------------------------------------------------------------------
+ * AUTO-GENERATED CLASS - DO NOT EDIT MANUALLY - for any changes edit GenerateArrowColumnSources and regenerate
+ * ---------------------------------------------------------------------------------------------------------------------
+ */
+package io.deephaven.extensions.arrow.sources;
+
+import io.deephaven.chunk.WritableChunk;
+import io.deephaven.chunk.WritableObjectChunk;
+import io.deephaven.chunk.attributes.Values;
+import io.deephaven.engine.rowset.RowSequence;
+import io.deephaven.engine.table.ChunkSource;
+import io.deephaven.engine.table.impl.ImmutableColumnSourceGetDefaults;
+import io.deephaven.extensions.arrow.ArrowWrapperTools;
+import org.apache.arrow.vector.BitVector;
+import org.apache.arrow.vector.types.pojo.Field;
+import org.jetbrains.annotations.NotNull;
+
+/**
+ * Arrow Vector: {@link BitVector}
+ * Deephaven Type: boolean
+ */
+public class ArrowBooleanColumnSource extends AbstractArrowColumnSource<Boolean> implements ImmutableColumnSourceGetDefaults.ForBoolean {
+    public ArrowBooleanColumnSource(final int highBit, final @NotNull Field field,
+            final ArrowWrapperTools. @NotNull Helper arrowHelper) {
+        super(boolean.class, highBit, field, arrowHelper);
+    }
+
+    @Override
+    public void fillChunk(final ChunkSource. @NotNull FillContext context,
+            final @NotNull WritableChunk<? super Values> destination,
+            final @NotNull RowSequence rowSequence) {
+        final WritableObjectChunk<Boolean, ? super Values> chunk = destination.asWritableObjectChunk();
+        final ArrowWrapperTools.FillContext arrowContext = (ArrowWrapperTools.FillContext) context;
+        chunk.setSize(0);
+        fillChunk(arrowContext, rowSequence, rowKey -> chunk.add(extract(getPositionInBlock(rowKey), arrowContext.getVector(field))));
+    }
+
+    @Override
+    public final Boolean get(final long rowKey) {
+        try (ArrowWrapperTools.FillContext fc = (ArrowWrapperTools.FillContext) makeFillContext(1)) {
+            fc.ensureLoadingBlock(getBlockNo(rowKey));
+            return extract(getPositionInBlock(rowKey), fc.getVector(field));
+        }
+    }
+
+    private Boolean extract(final int posInBlock, final BitVector vector) {
+        return vector.getObject(posInBlock);
+    }
```

Comment:
Could we do better by doing bulk copies of the vector and validity buffer and reconciling nulls ourselves afterwards? (I'm not sure.)

Reply:
This ends up being a wash due to the need to reconcile nulls. The non-direct-buffer copying version is simpler and will be easier to maintain. I've filed #3633 which would load data directly into a page cache via Chunks (via Barrage code paths) and is more likely to yield better performance.